frontend/components/commons/forms/questions-form/QuestionsForm.component.vue

```diff
@@ -110,8 +136,66 @@ export default {
         return input;
       });
     },
-    onSubmit() {
-      console.log(this.inputs);
+    async onSubmit() {
+      const createOrUpdateResponse = isResponsesByUserIdExists(
+        this.userId,
+        this.recordId
+      )
+        ? STATUS_RESPONSE.UPDATE
+        : STATUS_RESPONSE.CREATE;
+
+      const isSomeRequiredQuestionHaveNoAnswer = this.inputs.some(
+        (input) =>
+          input.is_required &&
+          input.options.every(
+            (option) => !option.value || option.value.length === 0
+          )
+      );
+      if (isSomeRequiredQuestionHaveNoAnswer) {
+        this.isError = true;
+        return;
+      }
+
+      let formattedSelectionOptionObject = {};
+      this.inputs.forEach((input) => {
+        // NOTE - if there is a responseid for the input, means that it's an update. Otherwise it's a create
+
+        let selectedOption = null;
+        switch (input.component_type) {
+          case COMPONENT_TYPE.SINGLE_LABEL:
+          case COMPONENT_TYPE.RATING:
+            selectedOption = input.options?.find((option) => option.value);
```

Comment:
If we're building components for each kind of component, maybe makes sense to ask each component the selected option.
```suggestion
            selectedOption = input.options?.find((option) => option.value);
```
```suggestion
            selectedOption = input.selectedOption
```
I see here a complement point here if we start growing the component catalog.